### PR TITLE
Fix uploading snap to multiple channels

### PR DESCRIPTION
**Pull request by FirelightFlagboy**
Currently only the latest args `--release` is used by snapcraft for upload.

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -126,10 +126,15 @@ jobs:
...
-        run: snapcraft upload $(cat ${{ runner.temp }}/snap-releases-args) snap/Parsec_${{ steps.version.outputs.full }}_linux_*.snap
+        run: |
+          set -x
+          for channel_arg in $(cat ${{ runner.temp }}/snap-releases-args); do
+            snapcraft upload $channel_arg snap/Parsec_${{ steps.version.outputs.full }}_linux_*.snap
+          done
         env:
           SNAPCRAFT_STORE_CREDENTIALS: ${{ secrets.SNAPCRAFT_CREDENTIALS }}
-        timeout-minutes: 5
+        # Each upload take ~2min, we could have up to 4 channel to upload so: `2 * 4 = 8min` (plus a little spare)
```
Comment (mmmarcos) [inline .github/workflows/publish.yml:136]:
```suggestion
        # Each upload takes ~2min, we could have up to 4 channels to upload so: `2 * 4 = 8min` (plus a little spare)
```

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -126,10 +126,15 @@ jobs:
         timeout-minutes: 2
 
       - name: Upload Snap
-        run: snapcraft upload $(cat ${{ runner.temp }}/snap-releases-args) snap/Parsec_${{ steps.version.outputs.full }}_linux_*.snap
+        run: |
+          set -x
+          for channel_arg in $(cat ${{ runner.temp }}/snap-releases-args); do
+            snapcraft upload $channel_arg snap/Parsec_${{ steps.version.outputs.full }}_linux_*.snap
```
Comment (mmmarcos) [inline .github/workflows/publish.yml:132]: Now that `snapcraft upload` is called 1 time for each channel, I think is clearer if we extract the `--release=` part _out_ of the `misc/snapcraft_releases.py` script. 
The script will therefore print only the channels to be updated, and this line will be: 

```suggestion
          for channel_arg in $(cat ${{ runner.temp }}/snap-releases-args); do
            snapcraft upload --release=$channel_arg snap/Parsec_${{ steps.version.outputs.full }}_linux_*.snap
```

Comment (FirelightFlagboy) [reply .github/workflows/publish.yml]: done

File: .github/workflows/publish.yml [hunk 3]
```diff
@@ -126,10 +126,15 @@ jobs:
         timeout-minutes: 2
 
       - name: Upload Snap
-        run: snapcraft upload $(cat ${{ runner.temp }}/snap-releases-args) snap/Parsec_${{ steps.version.outputs.full }}_linux_*.snap
+        run: |
+          set -x
+          for channel in $(cat ${{ runner.temp }}/snap-release-channels); do
+            snapcraft upload --releases=$channel snap/Parsec_${{ steps.version.outputs.full }}_linux_*.snap
```
Comment (mmmarcos) [inline .github/workflows/publish.yml:132]:
```suggestion
            snapcraft upload --release=$channel snap/Parsec_${{ steps.version.outputs.full }}_linux_*.snap
```

File: misc/snapcraft_releases.py [hunk 4]
```diff
@@ -1,6 +1,6 @@
 # Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS
 """
-For a given Parsec version, this script outputs the `--release=<channel>` options intended for snapcraft.
+For a given Parsec version, this script outputs the `<channel>` value intended to be used in the `--releases=<channel>` option for `snapcraft upload`.
```
Comment (mmmarcos) [inline misc/snapcraft_releases.py:3]: Approved too quickly ^^
```suggestion
For a given Parsec version, this script outputs the `<channel>` value intended to be used in the `--release=<channel>` option for `snapcraft upload`.
```

**Pull request merged**